packages/flutter/test/material/search_test.dart

```diff
@@ -40,6 +40,26 @@ void main() {
     TestDefaultBinaryMessengerBinding.instance!.defaultBinaryMessenger.setMockMethodCallHandler(SystemChannels.platform, null);
   });
 
+  testWidgets('Changing query moves cursor to the end of query', (
+      WidgetTester tester) async {
+    final _TestSearchDelegate delegate = _TestSearchDelegate();
+
+    await tester.pumpWidget(TestHomePage(delegate: delegate));
+    await tester.tap(find.byTooltip('Search'));
+    await tester.pumpAndSettle();
+
+    delegate.query = 'Foo';
+
+    final TextField textField =
+    tester.widget<TextField>(find.byType(TextField));
+
+    expect(
+        textField.controller!.selection,
+        TextSelection(
+            baseOffset: delegate.query.length,
+            extentOffset: delegate.query.length));
```

Comment:
indentation here is a bit dubious too, i recommend putting commas after the last `length` and the last penultimate `)` and applying the formatter to this statement to see the expected indentation.